server/server.js

```diff
@@ -20,8 +25,74 @@ app.use(express.static(__dirname + '/../public/'));
 // app.get('/api/acct/signedin', acctController.checkAuth);
 // app.get('/api/acct/logout', acctController.checkAuth);
 // app.post('/api/acct/addToList', acctController.addToList);
+
+
+
+
+
+// configuration ===========================================
+
+// Passport Facebook strategy configuration=================
+
+// OAuth 2.0-based strategies require a `verify` function which receives the
+// credential (`accessToken`) for accessing the Facebook API on the user's
+// behalf, along with the user's profile.  The function must invoke `cb`
+// with a user object, which will be set at `req.user` in route handlers after
+// authentication.
+var userController = require('./users/userController');
+
+passport.use(new Strategy({
+    clientID: auth.clientID,
+    clientSecret: auth.clientSecret,
+    callbackURL: auth.callbackURL,
+    profileFields: ['id', 'displayName', 'picture.height(150).width(150)','friends']
+  },
+  function(accessToken, refreshToken, profile, cb) {
+    //call a function which checks if user is in db
+    userController.createOrFindOne(profile);
+    return cb(null, profile);
```

Comment:
Why do we return the result of this callback execution instead of just invoking it? Also, since this is a node-style callback it should be able to handle errors with the first argument of the callback.